File: Advent of Code/2022/day 1/day1p1.hs
import qualified System.IO (isEOF)

solve :: [[Int]] -> IO ()
solve elves =
  System.IO.isEOF >>= \isEOF ->
    if isEOF
      then
        print
          ( maximum
              (map sum elves)
          )
      else
        getLine >>= \line ->
          if line == ""
            then solve ([] : elves)
            else
              let head : rest = elves
               in solve ((read line : head) : rest)

main :: IO ()
main = solve [[]]

File: Advent of Code/2022/day 1/p1/day1p1.erl
-module(day1p1).
-export([main/0]).

solve(Elves) ->
    case io:get_line('') of
        eof ->
            io:fwrite("~B", [lists:max(lists:map(fun lists:sum/1, Elves))]);
        UnstrippedLine ->
            Line = string:trim(UnstrippedLine, trailing, "\n"),
            if
                Line == "" ->
                    solve([[] | Elves]);
                true ->
                    [Head | Rest] = Elves,
                    solve([[list_to_integer(Line) | Head] | Rest])
            end
    end.

main() ->
    solve([[]]).
